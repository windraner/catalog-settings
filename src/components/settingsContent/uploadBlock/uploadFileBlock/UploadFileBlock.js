import React from 'react';
import Title from '../../title/Title';
import Hint from '../../hint/Hint';
import LinkedHint from '../linkedHint/LinkedHint';
import UploadButton from './uploadButton/UploadButton';
import PropTypes from 'prop-types';

const UploadFileBlock = (props) => {
  return (
    <div>
      <Title title={'Direct upload of products CSV file'} />

      <Hint text={'Here you can bulk upload products data using a products CSV file.'}
      />

      <div className="margin-top-20">
        <UploadButton
          {...props}
        />
      </div>

      <div className="margin-top-20">
        <LinkedHint
          link={'#'}
        />
      </div>
    </div>
  );
};

UploadFileBlock.propTypes = {
  inputHandler: PropTypes.func.isRequired,
  fileUrl: PropTypes.string.isRequired,
  file: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]).isRequired,
  flags: PropTypes.object,
};

export default UploadFileBlock;
