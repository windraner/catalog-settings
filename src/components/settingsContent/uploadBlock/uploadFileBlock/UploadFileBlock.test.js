import React from 'react';
import UploadFileBlock from './UploadFileBlock';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<UploadFileBlock />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<UploadFileBlock
      inputHandler={function() {}}
      fileUrl=""
      file=""
      flags={{}}
    />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
