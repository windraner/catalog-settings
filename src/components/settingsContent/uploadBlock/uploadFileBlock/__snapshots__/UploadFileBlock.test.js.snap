// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadFileBlock /> matches the snapshot 1`] = `
<div>
  <Title
    title="Direct upload of products CSV file"
  />
  <Hint
    text="Here you can bulk upload products data using a products CSV file."
  />
  <div
    className="margin-top-20"
  >
    <UploadButton
      file=""
      fileUrl=""
      flags={Object {}}
      inputHandler={[Function]}
    />
  </div>
  <div
    className="margin-top-20"
  >
    <LinkedHint
      link="#"
    />
  </div>
</div>
`;
